addons/mission/functions/fnc_log.sqf

```diff
@@ -0,0 +1,32 @@
+#include "script_component.hpp"
+/*
+ * Author: Mike
+ * Function based logging for 3DEN / Server
+ * Call from Functions
+ *
+ * Shows in-game warning and logs to RPT, optional logging to server/client RPT
+ *
+ * Arguments:
+ * 0: Function Name <STRING>
+ * 1: Error Message <STRING>
+ * 2: Log to Server <BOOL> (default: false)
+ *
+ * Return Value:
+ * None
+ *
+ * Examples:
+ * ["[TAC Mission X]:", format ["Undefined value %1", _value]] call FUNC(log)
```

Comment:
This is too verbose.
- Name should just be `Mission X`, because they are all `TAC` and that can be added in this function.
- Name should just use `__FILE__` preprocessor command.

This function should just be coupled with a macro to make the logging call as short as possible, eg. `MLOG("Undefined value %1",_value);` (and `MLOG_RPT` similar).